### PR TITLE
feat: machine readable record feedback JSON schema

### PR DESCRIPTION
**Pull request by andrewpollock**
First iteration of the JSON schema for encapsulating machine-readable feedback on individual OSV records.

Part of #2189

### CODE REVIEW

File: osv/feedback.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                }]
+            ]
+        }
+    },
+    "examples": [{
+        "invalid_records": [{
+            "id": "GSD-2021-1001229",
+            "finding_first_seen": "2024-04-01T08:15:00Z",
+            "import_last_attempt": "2024-05-03T08:15:00Z",
+            "findings": "INVALID_JSON"
```
Comment (another-rex) [inline osv/feedback.json:90]: These examples seem invalid, this should be an array right?

File: osv/feedback.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                        "type": "string",
+                        "title": "The id of the OSV record",
+                        "examples": [
+                            "GSD-2021-1001229",
+                            "GSD-2022-1003535"
+                        ]
+                    },
+                    "import_last_attempt": {
+                        "type": "string",
+                        "title": "When an import for this record was last attempted in RFC3339",
```
Comment (another-rex) [inline osv/feedback.json:35]: nit: Should we use ISO8601 instead?, it seems to be the more popular format from what I understand. (I think they are both very similar)

e.g. python has a toisoformat and fromisoformat which uses 8601. but I know golang only has RFC3339.

Apparently neither is a superset of the other...  https://ijmacd.github.io/rfc3339-iso8601/

File: osv/feedback.json (hunk 1 above)
Comment (andrewpollock) [reply osv/feedback.json]: Yes, I did an incomplete fix up of this late breaking change. Corrected in [8a7ca49](https://github.com/google/osv.dev/pull/2541/commits/8a7ca49d5dd223558f5d9a06fa3b797497f46936)

File: osv/feedback.json (hunk 2 above)
Comment (andrewpollock) [reply osv/feedback.json:35]: Yes, RFC3339 is indeed an annoying variant. https://xkcd.com/927/
I went with RFC3339 for consistency with the rest of the OSV Schema.

**Review by another-rex — APPROVED** [review]

**Pull request merged**